File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/domain/customer/valueobjects/Address.java
package br.com.deltaglobalbank.customers.domain.customer.valueobjects;

import java.util.Objects;

import br.com.deltaglobalbank.customers.domain.shared.valueobjects.Uf;

public record Address(
    String cep,
    String street,
    String city,
    Uf state,
    String country,
    String number,
    String complement,
    String neighborhood
) {
    public Address {
        Objects.requireNonNull(cep, "address_cep_required");
        Objects.requireNonNull(street, "address_street_required");
        Objects.requireNonNull(city, "address_city_required");
        Objects.requireNonNull(state, "address_state_required");
        if (!cep.matches("^\\d{8}$")) {
            throw new IllegalArgumentException("address_cep_invalid");
        }
        if (street.isBlank()) {
            throw new IllegalArgumentException("address_street_blank");
        }
        if (city.isBlank()) {
            throw new IllegalArgumentException("address_city_blank");
        }
    }

    public Address(String cep, String street, String city, Uf state) {
        this(cep, street, city, state, "BR", null, null, null);
    }
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/domain/customer/valueobjects/Address.kt
package br.com.deltaglobalbank.customers.domain.customer.valueobjects

import br.com.deltaglobalbank.customers.domain.shared.valueobjects.Uf

data class Address(
    val cep: String,
    val street: String,
    val city: String,
    val state: Uf,
    val country: String = "BR",
    val number: String? = null,
    val complement: String? = null,
    val neighborhood: String? = null,
) {
    init {
        require(CEP.matches(cep)) { "address_cep_invalid" }
        require(street.isNotBlank()) { "address_street_blank" }
        require(city.isNotBlank()) { "address_city_blank" }
    }
    companion object {
        private val CEP = Regex("^\\d{8}$")
    }
}